### PR TITLE
Added Job for converting a Bliss lexicon to plain text form

### PR DESCRIPTION
**Pull request by mattiadg**
The current version prints only the words for which at least one pronunciation exists and takes only the first pronunciation.

### CODE REVIEW

File: lexicon/conversion.py [hunk 1]
```diff
@@ -41,6 +41,33 @@ def run(self):
...
+        self.bliss_lexicon = bliss_lexicon
+        self.apply_filter = apply_filter
+
+        self.out_lexicon = self.output_path("pron.txt")
+
+    def tasks(self):
+        yield Task("run", mini_task=True)
+
+    def run(self):
+        mode = "rb" if self.bliss_lexicon.get_path().endswith(".gz") else "r"
```
Comment (curufinwe) [inline lexicon/conversion.py:57]: why "rb" for .gz files?

File: lexicon/conversion.py [hunk 2]
```diff
@@ -41,6 +41,33 @@ def run(self):
...
+        with uopen(tk.uncached_path(self.bliss_lexicon), mode) as lexicon_file:
```
Comment (curufinwe) [inline lexicon/conversion.py:58]: tk.uncached_path is not needed here (yes the code above is also not perfect)

File: lexicon/conversion.py [hunk 3]
```diff
@@ -41,6 +41,33 @@ def run(self):
...
+            lexicon = ET.fromstring(lexicon_file.read())
+            words = dict()
+            for e in lexicon.findall("lemma"):
+                orth = e.find("orth")
+                phon = e.findall("phon")
+                if orth.text is not None and len(phon) > 0:
+                    words[orth.text] = phon[0].text
+        with open(self.out_lexicon.get_path(), "w") as word_file:
```
Comment (curufinwe) [inline lexicon/conversion.py:66]: newline before this line would look nicer

File: lexicon/conversion.py [hunk 4]
```diff
@@ -41,6 +41,33 @@ def run(self):
...
+    def run(self):
+        mode = "rb" if self.bliss_lexicon.get_path().endswith(".gz") else "r"
+        with uopen(tk.uncached_path(self.bliss_lexicon), mode) as lexicon_file:
+            lexicon = ET.fromstring(lexicon_file.read())
+            words = dict()
+            for e in lexicon.findall("lemma"):
+                orth = e.find("orth")
+                phon = e.findall("phon")
+                if orth.text is not None and len(phon) > 0:
+                    words[orth.text] = phon[0].text
```
Comment (curufinwe) [inline lexicon/conversion.py:65]: here you also just print the first phoneme seq. Your commit description sounds like you want to have all pronunciations?

**Review by curufinwe — CHANGES_REQUESTED** [review]

**Comment by michelwi** [comment]
A similar job already exists here
https://github.com/rwth-i6/i6_core/blob/e38ad385e4b257aa0db6c22056b737cb37a8f07d/g2p/convert.py#L14

Check if it can be reused / extended for your purposes, or if your job can replace the other one.

File: lexicon/conversion.py [hunk 5]
```diff
@@ -41,6 +41,33 @@ def run(self):
                 word_file.write("%s\n" % w)
 
 
+class LexiconToTextListJob(Job):
+    def __init__(self, bliss_lexicon, apply_filter=True):
+        self.set_vis_name("Lexicon to Text Pair List")
+
+        self.bliss_lexicon = bliss_lexicon
+        self.apply_filter = apply_filter
```
Comment (michelwi) [inline lexicon/conversion.py:49]: `apply_filter` never used?

**Comment by mattiadg** [comment]
> A similar job already exists here
> 
> https://github.com/rwth-i6/i6_core/blob/e38ad385e4b257aa0db6c22056b737cb37a8f07d/g2p/convert.py#L14
> 
> Check if it can be reused / extended for your purposes, or if your job can replace the other one.

I didn't find this class, but it looks exactly what I needed.

**Pull request closed without merging**